src/lib/ext-monitor.js

```diff
@@ -0,0 +1,91 @@
+export default class ExtensionMonitor {
+  logs = [];
+  extensionMapList = new Map([]);
+
+  async isExtensionPageOpen() {
+    const tab = await browser.tabs.query({
+      url: browser.runtime.getURL('activitylog/activitylog.html'),
+    });
+    if (tab.length) return true;
+    return false;
+  }
+  // If the extension page is open, each logs will be send to
+  // extension page (as soon as it is encountered), if it is open.
+  async sendLogs(details) {
+    const isExtPageOpen = await this.isExtensionPageOpen();
+    if (isExtPageOpen) {
+      await browser.runtime.sendMessage({
+        updateLogs: details,
+      });
+    }
+  }
+
+  logListener() {
+    const listener = async (details) => {
+      // Everytime a log is encountered,
+      // it is being pushed to logs array.
+      this.logs.push(details);
+      //it checks if the extension page is open so that it can send log
+      await this.sendLogs(details);
```

Comment:
Here, I am waiting for the promise to be resolved, it is meant to resolve when the extension page is opened. So, it is require to handle the error if promise is reject from [`sendLogs()`](https://github.com/mozilla/extension-activity-monitor/pull/3/files#diff-aa663d2fedefb755dd0cf06fef1f44e2R18) method ?

Reply:
If anything, I suggest to catch (expected) errors in the `sendLogs` method itself instead of a catch-all in the caller of sendLogs.

In normal circumstances (= tab is open and listening for events) there shouldn't be any error, so I wouldn't catch anything right now.